### DISCUSSION TITLE
Array element value changes and DataChangeMonitoringFilter

### DISCUSSION THREAD

**Discussion by periclestsellos** [eclipse-milo/milo#1144] [General]
When we put an array on the opcua server, we may have an instance where an individual element of the array changes, rather than the entire array itself.

The DataChangeMonitoringFilter uses `Variant.equals` to check for value changes, but in the case of an element change in an array, this does not detect the change (because its not comparing two different arrays, the array reference is the same before and after).

Is there a way to manually notify the server of a Variant value change in order to trigger a subscription publish?

**Comment by kevinherron** [comment]
No, you should treat arrays as immutable, and create a new Variant containing a new array instance.

I'll give some thought to whether this could be changed in the 1.0 branch but I don't think there's anything else you can do for now.
